builtin/credential/okta/backend.go

```diff
@@ -106,6 +119,71 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		}
 		return nil, logical.ErrorResponse("okta authentication failed"), nil, nil
 
+	case "MFA_ENROLL", "MFA_ENROLL_ACTIVATE":
+		if b.Logger().IsDebug() {
+			b.Logger().Debug("auth/okta: user must enroll or complete mfa enrollment", "user", username)
+		}
+		return nil, logical.ErrorResponse("okta authentication failed: you must complete MFA enrollment to continue"), nil, nil
+
+	case "MFA_REQUIRED":
+		factorAvailable := false
+
+		var selectedFactor mfaFactor
+		// only okta push is currently supported
+		for _, v := range result.Embedded.Factors {
+			if v.Type == "push" {
+				factorAvailable = true
+				selectedFactor = v
+			}
+		}
+
+		if factorAvailable {
```

Comment:
Rather than put this whole block behind an if statement, do an `if !factorAvailable` and return an error immediately.